pkg/controller/servicebindingrequest/binder.go

```diff
@@ -211,21 +203,31 @@ func (b *binder) removeVolumes(volumes []interface{}) []interface{} {
 
 // extractSpecContainers search for
 func (b *binder) extractSpecContainers(obj *unstructured.Unstructured) ([]interface{}, error) {
-	log := b.logger.WithValues("Containers.NestedPath", containersPath)
+	log := b.logger.WithValues("Containers.NestedPath", b.getContainersPath())
 
-	containers, found, err := unstructured.NestedSlice(obj.Object, containersPath...)
+	containers, found, err := unstructured.NestedSlice(obj.Object, b.getContainersPath()...)
 	if err != nil {
 		return nil, err
 	}
 	if !found {
-		err = fmt.Errorf("unable to find '%#v' in object kind '%s'", containersPath, obj.GetKind())
+		err = fmt.Errorf("unable to find '%#v' in object kind '%s'", b.getContainersPath(), obj.GetKind())
 		log.Error(err, "is this definition supported by this operator?")
 		return nil, err
 	}
-
 	return containers, nil
 }
 
+// updateSecretField extract the specific secret field from
+// the object, and triggers an update.
+func (b *binder) updateSecretField(
+	obj *unstructured.Unstructured,
+) (*unstructured.Unstructured, error) {
+	if err := unstructured.SetNestedField(obj.Object, b.sbr.GetName(), b.getSecretFieldPath()...); err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
 // updateSpecContainers extract containers from object, and trigger update.
 func (b *binder) updateSpecContainers(
```

Comment:
I have created a separate task for this: https://github.com/redhat-developer/service-binding-operator/issues/601

Reply:
Before this PR, I admit this method's signature isn't the most straight-forward but it didn't have any side effects related to the given `obj`, but a side effect has been introduced by this PR since a mutation has been introduced here, so I find it appropriate to solve it in this PR.

Reply:
Let me know if this fix is fine. 669c7cc5dad963f6cae0dc91382ba74768e392c2

Reply:
See https://github.com/redhat-developer/service-binding-operator/pull/513/files#r469825678

Reply:
see the above link for the reply.